pkg/server/slicerouter_config.go

```diff
@@ -506,23 +432,53 @@ func sliceRouterInjectRoute(remoteSubnet string, nextHopIPList []string) error {
 
 		logger.GlobalLogger.Infof("RT reconciled at: %v", lastRoutingTableReconcileTime)
 	}
+
 	logger.GlobalLogger.Infof("sliceRouterInjectRoute", "remoteSubnetRouteMap", remoteSubnetRouteMap)
 
-	_, routePresent := remoteSubnetRouteMap.Load(remoteSubnet)
-	nextHopInfoSlice, err := getNextHopInfoSlice(nextHopIPList)
-	if err != nil {
-		return err
+	if len(nextHopIPList) == 0 {
+		// Treat this as a signal to delete the route to the remoteSubnet
+		err := sliceRouterDeleteRouteToDst(remoteSubnet)
+		if err != nil {
+			return err
+		}
+		remoteSubnetRouteMap.Delete(remoteSubnet)
+		return nil
 	}
 
-	for i := 0; i < len(nextHopIPList); i++ {
+	installRoute := false
 
-		nextHopList, _ := remoteSubnetRouteMap.Load(remoteSubnet)
-		if routePresent && checkRouteAdd(nextHopList.([]string), nextHopIPList[i]) {
-			logger.GlobalLogger.Infof("Ignoring route add request. Route already installed. RemoteSubnet: %v, NextHop: %v",
-				remoteSubnet, nextHopIPList[i])
-			continue
+	cachedNextHopValue, routePresent := remoteSubnetRouteMap.Load(remoteSubnet)
+	if !routePresent {
+		installRoute = true
+	} else {
+		cachedNextHopList := cachedNextHopValue.([]string)
+		// Route is present in the cache. Check if the stored nexthop matches with the nexthop received
+		// in the input param.
+		// We reinstall the route if the two lists do not match.
+		if len(cachedNextHopList) != len(nextHopIPList) {
+			installRoute = true
+		} else {
+			for _, nextHopInCache := range cachedNextHopList {
```

Comment:
shouldn't we loop on the `nextHopIPList` list, any new entries received in nextHopIPList are not getting checked!